### PR TITLE
Feature/like management

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
## ❓What is the change ❓

Add a like feature on the videos users can now like the videos only on the single video page.
Will think of  adding a like in menu component later,

## What is it ❓
- [ ] bug🐛
- [x] feature 👩‍💻
- [ ] fix 🔧

### CODE REVIEW

**Comment by AnkurChunekar** [comment]
- Once the user logs in you can change the button text to `Logout`.

**Comment by AnkurChunekar** [comment]
kevin pr
Look into these errors!

File: src/components/LandscapeVideoCard/LandscapeVideoCard.jsx [hunk 1]
```diff
@@ -18,9 +20,20 @@ function LandscapeVideoCard({
...
   const { historyDispatch } = useHistory();
+  const { likeDispatch } = useLike();
+  const deleteFromList = () => {
+    if (!!inHistory) {
+      deleteFromHistory(_id, authState.token, historyDispatch);
+    }
+    if (!!inLikedVideo) {
+      deleteFromLiked(_id, authState.token, likeDispatch);
+    }
+  };
```
Comment (AnkurChunekar) [inline src/components/LandscapeVideoCard/LandscapeVideoCard.jsx:36]: Just curious, `(!!inLikedVideo)` why have you used the Logical Not operator in this way?

**Review by AnkurChunekar — APPROVED** [review]
- Rest everything looks good to me.

**Comment by Kevin-Solomon** [comment]
> * Once the user logs in you can change the button text to `Logout`.

I will raise a PR for logout feature.

File: src/components/LandscapeVideoCard/LandscapeVideoCard.jsx (hunk 1 above)
Comment (Kevin-Solomon) [reply src/components/LandscapeVideoCard/LandscapeVideoCard.jsx:36]: It is actually redundant as it coerces to a boolean.
I will remove it

**Comment by Kevin-Solomon** [comment]
> kevin pr Look into these errors!

These are just failed network requests.
As they come in the catch block before pushing code to production I will remove them.

File: src/components/Drawer/Drawer.jsx [hunk 2]
```diff
@@ -22,8 +22,8 @@ function Drawer() {
       >
         <DrawerIcons name="History" type={'HISTORY'} size={'25px'} />
       </NavLink>
-      <NavLink to="/library">
-        <DrawerIcons name="Library" type={'LIBRARY'} size={'25px'} />
+      <NavLink to="/liked">
+        <DrawerIcons name="Liked Videos" type={'LIKE'} size={'25px'} />
       </NavLink>
       <NavLink to="/watch-layer">
```
Comment (kart-c) [inline src/components/Drawer/Drawer.jsx]: there is a typo here, also the route is not defined in `App.js`

**Review by kart-c — APPROVED** [review]
Functionality is working fine @Kevin-Solomon. I have a few suggestions

- Add `alt` attribute to `img` tags as this helps in accessibility 
- Since you have implemented auth, I would suggest making the "likes" page a private route. Liking a video can be restricted to a logged-in user.

**Comment by Kevin-Solomon** [comment]
> * Add `alt` attribute to `img` tags as this helps in accessibility
> * Since you have implemented auth, I would suggest making the "likes" page a private route. Liking a video can be restricted to a logged-in user.

I will raise PR for RequireAuth component soon.

File: src/components/Drawer/Drawer.jsx (hunk 2 above)
Comment (Kevin-Solomon) [reply src/components/Drawer/Drawer.jsx]: > there is a typo here, also the route is not defined in `App.js`

I have yet to add a watch later feature on the app.

**Pull request merged**